packages/lit-html/src/directives/cache.ts

```diff
@@ -41,68 +105,10 @@ import {
  * `
  * ```
  */
-export const cache = directive(
-  class extends Directive {
-    templateCache = new WeakMap<TemplateStringsArray, ChildPart>();
-    value?: TemplateResult;
+export const cache = directive(CacheDirectiveClass);
 
-    constructor(partInfo: PartInfo) {
-      super(partInfo);
-    }
-
-    render(v: unknown) {
-      // Return an array of the value to induce lit-html to create a ChildPart
-      // for the value that we can move into the cache.
-      return [v];
-    }
-
-    update(containerPart: ChildPart, [v]: DirectiveParameters<this>) {
-      // If the previous value is a TemplateResult and the new value is not,
-      // or is a different Template as the previous value, move the child part
-      // into the cache.
-      if (
-        isTemplateResult(this.value) &&
-        (!isTemplateResult(v) || this.value.strings !== v.strings)
-      ) {
-        // This is always an array because we return [v] in render()
-        const partValue = getCommittedValue(containerPart) as Array<ChildPart>;
-        const childPart = partValue.pop()!;
-        let cachedContainerPart = this.templateCache.get(this.value.strings);
-        if (cachedContainerPart === undefined) {
-          const fragment = document.createDocumentFragment();
-          cachedContainerPart = render(nothing, fragment);
-          this.templateCache.set(this.value.strings, cachedContainerPart);
-        }
-        // Move into cache
-        setCommittedValue(cachedContainerPart, [childPart]);
-        insertPart(cachedContainerPart, undefined, childPart);
-        childPart.setConnected(false);
-      }
-      // If the new value is a TemplateResult and the previous value is not,
-      // or is a different Template as the previous value, restore the child
-      // part from the cache.
-      if (
-        isTemplateResult(v) &&
-        (!isTemplateResult(this.value) || this.value.strings !== v.strings)
-      ) {
-        const cachedContainerPart = this.templateCache.get(v.strings);
-        if (cachedContainerPart !== undefined) {
-          // Move the cached part back into the container part value
-          const partValue = getCommittedValue(
-            cachedContainerPart
-          ) as Array<ChildPart>;
-          const cachedPart = partValue.pop()!;
-          // Move cached part back into DOM
-          clearPart(containerPart);
-          insertPart(containerPart, undefined, cachedPart);
-          setCommittedValue(containerPart, [cachedPart]);
-          cachedPart.setConnected(true);
-        }
-        this.value = v;
-      } else {
-        this.value = undefined;
-      }
-      return this.render(v);
-    }
-  }
-);
+/**
+ * Non-callable type of the directive class. Necessary for when a function or
+ * method returns the return type of the above directive.
+ */
+export type {CacheDirectiveClass};
```

Comment:
Here and elsewhere, naming a class with `Class` in it is awkward, and this comment is a bit confusing. Consider:

```typescript
/**
 * The type of the class that powers this directive. Necessary for naming its return type.
 */
export type {CacheDirective as CacheDirectiveType};
```